sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetRelation.scala

```diff
@@ -172,9 +173,14 @@ private[sql] object ParquetRelation {
       sqlContext.conf.parquetCompressionCodec.toUpperCase, CompressionCodecName.UNCOMPRESSED)
       .name())
     ParquetRelation.enableLogForwarding()
-    ParquetTypesConverter.writeMetaData(attributes, path, conf)
+    // This is a hack. We always set nullable/containsNull/valueContainsNull to true
+    // for the schema of a parquet data.
+    val schema =
+      DataType.alwaysNullable(StructType.fromAttributes(attributes)).asInstanceOf[StructType]
```

Comment:
I am not sure if we want to do this. For `createEmpty` (this change), we are creating a new parquet dataset. But, for ParquetRelation2, the schema may be created for an existing dataset. For every place we create a new parquet dataset, we will call alwaysNullable.